src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy

```diff
@@ -1,23 +1,31 @@
 package com.vherasymenko.avro.decoder.core
 
 import com.vherasymenko.avro.decoder.core.json.AvroJsonDecoderPort
-import event.course_install.CourseInstall
+import groovy.util.logging.Slf4j
+import org.apache.avro.Schema
+import org.springframework.cloud.stream.schema.client.SchemaRegistryClient
 
 /**
  * Decoder for the course install document.
  */
+@Slf4j
 class CourseInstallDecoderService implements CourseInstallDecoderPort {
 
-
     private final AvroJsonDecoderPort decoder
 
-    CourseInstallDecoderService( AvroJsonDecoderPort aDecoder ) {
+    private final SchemaRegistryClient registryClient
+
+    CourseInstallDecoderService( AvroJsonDecoderPort aDecoder, SchemaRegistryClient aRegistryClient ) {
         decoder = aDecoder
+        registryClient = aRegistryClient
     }
 
     @Override
-    void decodeCourseInstallEvent( String event ) {
-        def schema = CourseInstall.classSchema
-        decoder.decodeEvent( schema, event )
+    void decodeCourseInstallEvent( String event, int schemaId ) {
+        def writeSchema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        def readSchema = new Schema.Parser().parse( new File( 'src/main/resources/avro/course_install_2.avsc' ) )
```

Comment:
Just curious.  Why do get the `writeSchema` from the registry but the `readSchema` from the file system?